platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -0,0 +1,355 @@
+  @Override
+  public synchronized void run() {
+    if (!crlByUrlEnabled || crlLocationUrl == null) {
+      return;
+    }
+    if (!crlLocationUrl.startsWith(HTTPS)) {
+      postErrorEvent("The provided URL was not an HTTPS URL.");
+      return;
+    }
+    Object entity = getRemoteCrl();
+    if (!(entity instanceof InputStream)) {
+      postErrorEvent("Unable to download the CRL.");
+      return;
+    }
+    FileBackedOutputStream fileBackedOutputStream = null;
+    try {
+      // Read the response content and get the byte source
+      ByteSource byteSource;
+      try (InputStream entityStream = (InputStream) entity) {
+        fileBackedOutputStream = new FileBackedOutputStream(FILE_BACKED_STREAM_THRESHOLD);
+        IOUtils.copy(entityStream, fileBackedOutputStream);
+        fileBackedOutputStream.close();
+        byteSource = fileBackedOutputStream.asByteSource();
+      }
+      File crlFile = getCrlFile(byteSource);
+      // Verify that the CRL is valid
+      if (!crlIsValid(byteSource)) {
+        postErrorEvent("An error occurred while validating the CRL.");
+        return;
+      }
+      writeCrlToFile(byteSource, crlFile);
+    } catch (IOException e) {
+      LOGGER.warn("Unable to copy CRL to local CRL. {}", e.getMessage());
+      postErrorEvent("An error occurred while downloading the CRL.");
+    } finally {
+      // Cleanup temp file
+      if (fileBackedOutputStream != null) {
+        try {
+          fileBackedOutputStream.reset();
+        } catch (IOException e) {
+          LOGGER.warn("Error occurred while deleting the temporary file. {}", e.getMessage());
+        }
+      }
+    }
+  }
+  /**
+   * Get the CRL from the given URL.
+   *
+   * @return - the response body
+   */
+  private Object getRemoteCrl() {
+    SecureCxfClientFactory cxfClientFactory =
+        factory.getSecureCxfClientFactory(crlLocationUrl, WebClient.class);
+    WebClient client = cxfClientFactory.getWebClient();
+    Response response = client.get();
+    return response.getEntity();
+  }
+  /**
+   * Determines the CRL encoding and creates a CRL file.
+   *
+   * @param byteSource - CRL content as a byte source
+   * @return - the created file
+   * @throws IOException
+   */
+  private File getCrlFile(ByteSource byteSource) throws IOException {
+    // Determine the file extension
+    File crlFile;
+    try (InputStream inputStream = byteSource.slice(0, 300).openStream()) {
+      if (IOUtils.toString(inputStream, StandardCharsets.UTF_8).contains("-----BEGIN")) {
+        crlFile = new File(crlFileLocation + PEM_CRL);
+      } else {
+        crlFile = new File(crlFileLocation + DEM_CRL);
+      }
+    }
+    return crlFile;
+  }
+  /**
+   * Validates the given CRL by attempting to create a {@link CRL}
+   *
+   * @param byteSource - CRL byte source
+   * @return - True if the CRL is valid. False if its invalid
+   */
+  private boolean crlIsValid(ByteSource byteSource) {
```

Comment:
❗️ `InterruptedIOException` shouldn't be ignored. It should be left to propagate up so you can handle it as per my previous comment.